packages/nimble-components/src/banner/tests/banner.spec.ts

```diff
@@ -0,0 +1,125 @@
+async function setup(): Promise<Fixture<Banner>> {
+    return fixture<Banner>(html`
+        <nimble-banner>
+            <span slot="title">Title</span>
+            Message text
+        </nimble-banner>
+    `);
+}
+
+describe('Banner', () => {
+    let element: Banner;
+    let connect: () => Promise<void>;
+    let disconnect: () => Promise<void>;
+
+    /** A helper function to abstract adding a 'toggle' event listener, spying
+     * on the event being called, and removing the event listener. The returned promise
+     * should be resolved prior to completing a test.
+     */
+    function createToggleListener(): {
+        promise: Promise<void>,
+        spy: jasmine.Spy
+    } {
+        const spy = jasmine.createSpy();
+        return {
+            promise: new Promise(resolve => {
+                const handler = (...args: unknown[]): void => {
+                    element.removeEventListener('toggle', handler);
+                    spy(...args);
+                    resolve();
+                };
+                element.addEventListener('toggle', handler);
+            }),
+            spy
+        };
+    }
+
+    beforeEach(async () => {
+        ({ element, connect, disconnect } = await setup());
+        await connect();
+    });
+
+    afterEach(async () => {
+        await disconnect();
+    });
+
+    it('can construct an element instance', () => {
+        expect(document.createElement('nimble-banner')).toBeInstanceOf(Banner);
+    });
+
+    it("should initialize 'open' to false", () => {
+        expect(element.open).toBeFalse();
+    });
+
+    it("should be hidden when 'open' is false", async () => {
+        element.open = false;
+        await waitForUpdatesAsync();
+        expect(getComputedStyle(element).display).toBe('none');
+    });
+
+    it("should be visible when 'open' is true", async () => {
+        element.open = true;
+        await waitForUpdatesAsync();
+        expect(getComputedStyle(element).display).not.toBe('none');
+    });
+
+    it("should fire 'toggle' when 'open' is changed", async () => {
+        element.open = false;
+        const toggleListener = createToggleListener();
+        element.open = true;
+        await toggleListener.promise;
+        expect(toggleListener.spy).toHaveBeenCalledOnceWith(
+            new CustomEvent('toggle', {
+                detail: { oldState: false, newState: true }
+            })
+        );
+    });
+
+    it("should remove 'open' when dismiss button is clicked", () => {
+        element.open = true;
+        element.shadowRoot?.querySelector('nimble-button')?.click();
+        expect(element.open).toBeFalse();
+    });
+
+    it("should initialize 'severity' to default", () => {
+        expect(element.severity).toBe(BannerSeverity.default);
+    });
+
+    it("should hide title when 'titleHidden' set", async () => {
+        element.titleHidden = true;
+        await waitForUpdatesAsync();
+        expect(element.slottedTitle![0]).toBeUndefined();
```

Comment:
I think there is a problem here. If `slottedTitle` is empty when `titleHidden` is set, how are you getting an aria-label for the banner? Isn't the whole idea that the content in the slot still exists even when `titleHidden` is set?

It would be good to have a unit test that verifies that content within the `title` slot is still used as the label even when `titleHidden` is set.

edit: It actually looks like there is a testing gap regarding aria-label. I don't see any tests that verify that the label is set correctly in different scenarios.

Reply:
Great catch. Not sure how I missed that.

I couldn't figure out how to test that the title was actually visible or not, so I decided to make that part of our visual Storybook tests. I then created two tests to ensure the `aria-label` is set properly whether `title-hidden` is set or not.